system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/dataspaceconnector/test/e2e/EndToEndTransferTest.java

```diff
@@ -0,0 +1,343 @@
+
+import org.eclipse.dataspaceconnector.junit.launcher.EdcRuntimeExtension;
+import org.eclipse.dataspaceconnector.policy.model.Action;
+import org.eclipse.dataspaceconnector.policy.model.Permission;
+import org.eclipse.dataspaceconnector.policy.model.Policy;
+import org.eclipse.dataspaceconnector.policy.model.PolicyType;
+import org.eclipse.dataspaceconnector.spi.asset.AssetSelectorExpression;
+import org.eclipse.dataspaceconnector.spi.types.domain.DataAddress;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.negotiation.ContractOfferRequest;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractDefinition;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractOffer;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataRequest;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcess;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferType;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+import static java.io.File.separator;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.eclipse.dataspaceconnector.common.testfixtures.TestUtils.getFreePort;
+import static org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcessStates.COMPLETED;
+import static org.hamcrest.CoreMatchers.notNullValue;
+
+public class EndToEndTransferTest {
+
+    private final Duration timeout = Duration.ofSeconds(30);
+    private static final String API_KEY_CONTROL_AUTH = "password";
+
+    private static final URI CONSUMER_CONTROL_PLANE = URI.create("http://localhost:" + getFreePort());
+    private static final URI CONSUMER_DATA_PLANE = URI.create("http://localhost:" + getFreePort());
+    private static final URI CONSUMER_DATA_PLANE_PUBLIC = URI.create("http://localhost:" + getFreePort() + "/public");
+    private static final URI CONSUMER_BACKEND_SERVICE = URI.create("http://localhost:" + getFreePort());
+    private static final URI PROVIDER_CONTROL_PLANE = URI.create("http://localhost:" + getFreePort());
+    private static final URI PROVIDER_DATA_PLANE = URI.create("http://localhost:" + getFreePort());
+    private static final URI PROVIDER_DATA_PLANE_PUBLIC = URI.create("http://localhost:" + getFreePort() + "/public");
+    private static final URI PROVIDER_BACKEND_SERVICE = URI.create("http://localhost:" + getFreePort());
+
+    @RegisterExtension
+    static EdcRuntimeExtension consumerControlPlane = new EdcRuntimeExtension(
```

Comment:
nit: what's your feedback on the experience of using this `EdcRuntimeExtension` to develop end-to-end tests?

Reply:
@algattik It was really good especially because it gives you fast feedback and possibility to debug. Sometimes I forgot to rebuild the modules I was modifying so I didn't get the expected outcomes but since I get used too it became really smooth.